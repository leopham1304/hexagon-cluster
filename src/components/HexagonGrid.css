.tiles {
  fill: burlywood;
}

.name {
  fill: brown;
  font-size: 5px;
}