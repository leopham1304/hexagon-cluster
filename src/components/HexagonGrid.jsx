import React from "react";
import { HexGrid, Layout, Hexagon, Text } from "react-hexgrid";
import PropTypes from "prop-types";
import "./HexagonGrid.css";

const HexagonGrid = ({ hexagonList }) => (
  <HexGrid width="100%" height={1000} viewBox="-50 -50 100 100">
    <Layout
      className="tiles"
      size={{ x: 8, y: 8 }}
      flat={true}
      spacing={1.005}
      origin={{ x: 0, y: 0 }}
    >
      {hexagonList.map((hex, i) => (
        <Hexagon key={i} q={hex.x} r={hex.z} s={hex.y} data={hex}>
          <Text className="name">{hex.name}</Text>
        </Hexagon>
      ))}
    </Layout>
  </HexGrid>
);

HexagonGrid.propTypes = {
  hexagonList: PropTypes.PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default HexagonGrid;
