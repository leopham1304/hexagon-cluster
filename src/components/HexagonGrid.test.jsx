import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import HexagonGrid from "./HexagonGrid.jsx";

describe("HexagonGrid", () => {
  const wrapper = shallow(<HexagonGrid hexagonList={[{name: "test", x: 1, y:2, z:3}]} />);

  it("should render the HexagonGrid Component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
