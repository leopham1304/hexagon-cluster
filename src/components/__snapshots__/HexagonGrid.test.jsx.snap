// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexagonGrid should render the HexagonGrid Component correctly 1`] = `
<HexGrid
  height={1000}
  viewBox="-50 -50 100 100"
  width="100%"
>
  <Layout
    className="tiles"
    flat={true}
    origin={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    size={
      Object {
        "x": 8,
        "y": 8,
      }
    }
    spacing={1.005}
  >
    <Hexagon
      data={
        Object {
          "name": "test",
          "x": 1,
          "y": 2,
          "z": 3,
        }
      }
      key="0"
      q={1}
      r={3}
      s={2}
    >
      <Text
        className="name"
      >
        test
      </Text>
    </Hexagon>
  </Layout>
</HexGrid>
`;
